### PR TITLE
♻ #65835 Apply suggested feedback.

### PR DESCRIPTION
**Pull request by juandausa**
Internal review.

### CODE REVIEW

File: packages/file_selector/file_selector_windows/lib/file_selector_windows.dart [hunk 1]
```diff
@@ -3,23 +3,20 @@
...
 import 'src/messages.g.dart';
 
 /// An implementation of [FileSelectorPlatform] for Windows.
 class FileSelectorWindows extends FileSelectorPlatform {
-  /// Constructor for filePicker.
-  FileSelectorWindows([this._dartFileSelectorAPI]) {
-    _dartFileSelectorAPI = _dartFileSelectorAPI ?? DartFileSelectorAPI();
-    _api = _dartFileSelectorAPI!;
-  }
+  /// Constructor for FileSelectorWindows. It recieves a DartFileSelectorApi parameter allowing dependency injection.
```
Comment (gantelo) [inline packages/file_selector/file_selector_windows/lib/file_selector_windows.dart:11]:
```suggestion
  /// Constructor for FileSelectorWindows. It receives a DartFileSelectorApi parameter allowing dependency injection.
```

File: packages/file_selector/file_selector_windows/lib/file_selector_windows.dart [hunk 2]
```diff
@@ -3,23 +3,20 @@
...
+      : _api = api ?? FileSelector.withoutParameters();
 
-  late DartFileSelectorAPI _api;
-  late DartFileSelectorAPI? _dartFileSelectorAPI;
+  final FileSelector _api;
 
-  /// Registers the Windows implementation.
-  static void registerWith([DartFileSelectorAPI? filePicker]) {
-    FileSelectorPlatform.instance = FileSelectorWindows(filePicker);
+  /// Registers the Windows implementation. It recieves a DartFileSelectorApi parameter allowing dependency injection.
```
Comment (gantelo) [inline packages/file_selector/file_selector_windows/lib/file_selector_windows.dart:17]:
```suggestion
  /// Registers the Windows implementation. It receives a DartFileSelectorApi parameter allowing dependency injection.
```

File: packages/file_selector/file_selector_windows/lib/src/file_open_dialog_wrapper.dart [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  int setOptions(int options, IFileOpenDialog dialog) {
+    return dialog.setOptions(options);
+  }
+
+  /// Returns the IFileOpenDialog's [options](https://pub.dev/documentation/win32/latest/winrt/FILEOPENDIALOGOPTIONS-class.html).
+  int getOptions(Pointer<Uint32> ptrOptions, IFileOpenDialog dialog) {
+    return dialog.getOptions(ptrOptions);
+  }
+
+  /// Sets confirmation button text on an IFileOpenDialog. If the [confirmationText] is null 'Pick' will be used.
```
Comment (gantelo) [inline packages/file_selector/file_selector_windows/lib/src/file_open_dialog_wrapper.dart:21]:
```suggestion
  /// Sets confirmation button text on an IFileOpenDialog. If the [confirmationText] is null, 'Pick' will be used.
```

File: packages/file_selector/file_selector_windows/lib/src/file_open_dialog_wrapper.dart [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    return dialog.release();
+  }
+
+  /// Return a result from an IFileOpenDialog.
+  int getResult(
+      Pointer<Pointer<COMObject>> ptrCOMObject, IFileOpenDialog dialog) {
+    return dialog.getResult(ptrCOMObject);
+  }
+
+  /// Return results from an IFileOpenDialog, this should be used when selecting multiple items.
```
Comment (gantelo) [inline packages/file_selector/file_selector_windows/lib/src/file_open_dialog_wrapper.dart:65]:
```suggestion
  /// Return results from an IFileOpenDialog. This should be used when selecting multiple items.
```

File: packages/file_selector/file_selector_windows/lib/src/file_selector.dart [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+    }
+
+    if (selectionOptions.allowMultiple) {
+      options |= FILEOPENDIALOGOPTIONS.FOS_ALLOWMULTISELECT;
+    }
+
+    return options;
+  }
+
+  /// Sets the dilog options based on the fileMustExist value and the selectionOption given.
```
Comment (gantelo) [inline packages/file_selector/file_selector_windows/lib/src/file_selector.dart:147]:
```suggestion
  /// Sets the dialog options based on the fileMustExist value and the selectionOption given.
```

File: packages/file_selector/file_selector_windows/lib/src/file_selector.dart [hunk 6]
```diff
@@ -0,0 +1,395 @@
...
+
+      hResult = _fileOpenDialogWrapper.setFolder(ptrPath, dialog);
+
+      _validateResult(hResult);
+    });
+
+    return hResult;
+  }
+
+  /// Initilaize the COM library with the internal [CoInitializeEx](https://pub.dev/documentation/win32/latest/winrt/CoInitializeEx.html) method.
```
Comment (gantelo) [inline packages/file_selector/file_selector_windows/lib/src/file_selector.dart:186]:
```suggestion
  /// Initialize the COM library with the internal [CoInitializeEx](https://pub.dev/documentation/win32/latest/winrt/CoInitializeEx.html) method.
```

File: packages/file_selector/file_selector_windows/lib/src/file_selector.dart [hunk 7]
```diff
@@ -0,0 +1,395 @@
...
+  /// A [WindowsException] is thrown if an error occurs.
+  @visibleForTesting
+  int initializeComLibrary() {
+    final int hResult = _fileOpenDialogWrapper.coInitializeEx();
+    _validateResult(hResult);
+    return hResult;
+  }
+
+  /// Returns a list directory paths from user interaction. It recieves the IFileOpenDialog show result to verify wether the user has cancelled the dialog or not.
```
Comment (gantelo) [inline packages/file_selector/file_selector_windows/lib/src/file_selector.dart:195]:
```suggestion
  /// Returns a list of directory paths from user interaction. It receives the IFileOpenDialog show result to verify whether the user has canceled the dialog or not.
```

File: packages/file_selector/file_selector_windows/lib/src/file_selector.dart [hunk 8]
```diff
@@ -0,0 +1,395 @@
...
+      }
+    } else {
+      hResult = _getSelectedPathsFromUserInput(
+          selectionOptions, selectedElements, dialog);
+    }
+
+    return selectedElements;
+  }
+
+  /// Sets confirmation button text on an IFileOpenDialog. If the [confirmationText] is null 'Pick' will be used.
```
Comment (gantelo) [inline packages/file_selector/file_selector_windows/lib/src/file_selector.dart:214]:
```
/// Sets the confirmation button text on an IFileOpenDialog. If the [confirmationText] is null, 'Pick' will be used.
```

Comment (juandausa) [reply packages/file_selector/file_selector_windows/lib/src/file_selector.dart:214]: @gantelo what do you mean by this?

Comment (juandausa) [reply packages/file_selector/file_selector_windows/lib/src/file_selector.dart:214]: Thanks!

**Pull request merged**